tests/components/deconz/test_device_trigger.py

```diff
@@ -321,10 +311,16 @@ async def test_functional_device_trigger(
                 "uniqueid": "d0:cf:5e:ff:fe:71:a4:3a-01-1000",
             }
         }
-    }
-    with patch.dict(DECONZ_WEB_REQUEST, data):
-        await setup_deconz_integration(hass, aioclient_mock)
-
+    ],
+)
+@pytest.mark.usefixtures("config_entry_setup")
+async def test_functional_device_trigger(
+    hass: HomeAssistant,
+    device_registry: dr.DeviceRegistry,
+    service_calls: list[ServiceCall],
+    mock_deconz_websocket,
```

Comment:
If order doesn't matter here you could use `@pytest.mark.usefixtures` for `mock_deconz_websocket`

Reply:
Actually it looks like you call it, so would be nice to type it, but it looks like it only uses kwargs?

Reply:
Suggested type hint:
```python
@pytest.fixture(autouse=True)
def mock_deconz_websocket() -> Generator[Callable[..., Awaitable[None]]]:
```
and
```python
async def test_xxx(
    ...,
    mock_deconz_websocket: Callable[..., Awaitable[None]],
    ...,
) -> None:
```

Reply:
Thanks! I will do this in a standalone PR

Reply:
Improved websocket fixtures https://github.com/home-assistant/core/pull/121332